### DISCUSSION TITLE
Modular Weaponry System

### DISCUSSION THREAD

**Discussion by daPotato** [unitystation/unitystation#6987] [Ideas]
Ayy suggestion time babie.
With work beginning on a major sprite overhaul of the shooty shit in the game, I've pretty much decided that I want the guns to be modular.
Features would include attachable and detachable sights, barrel attachments, laser pointers, flashlights, stocks and picatinny foregrips.

More advanced modifications, less of a priority but would be great to have, could include grip swaps (swapping between a pistol grip and a rifle grip on compatible guns), integrated foregrip swaps (like the one piece wooden foregrips on some AK platform rifles), and receiver modifications. These advanced mods would require use of tools such as screwdrivers and would take longer overall.

As well as the advanced modifications, there could be ghetto mods, based on attaching shit to stuff that shouldn't accept it, such as welding a scope to a gun that doesn't have a rail, but resulting in an accuracy debuff.

Thots and opinions, questions, comments, concerns, ideas down below, the whole thing has some time to bake since I'm only just now starting on it.

**Comment by braincrab** [comment]
i love the idea of attaching mods to guns they really shouldn't be on. like putting a scope and silencer on a tazer.
